tools/nimgrep.nim

```diff
@@ -50,44 +58,132 @@ Options:
                -b:N   print N lines of leading context before every match
   --context:N, -c:N   print N lines of leading context before every match and
                       N lines of trailing context after it
+  --sortTime          order files by the last modification time -
```

Comment:
This needs more thought.

I can see how we'll quickly end up with a combinatorial explosion of flags and increased complexity / bugs. Instead we should use separation of concern (file listing vs file content matching) to reduce flags and complexity and increase flexibility as follows:

* support reading input files from stdin

This is an essential feature of `ripgrep` (arguably the best file searcher available eg see https://blog.burntsushi.net/ripgrep/ https://github.com/BurntSushi/ripgrep), eg:

```
find . | rg foo
find . -depth 2 | sort | rg foo
```
etc.

This allows pipelining and enables more flexibility than if you were to cram everything in a single program invocation

Reply:
Yes, its presence is questionable, I agree. And it's not a Unix style.

But telling people to learn Shell is also controversial advice.

Sorting by time from Shell is not really that easy:
`find . -type f -printf "%T@ %p\n" | sort -n | cut -f2 -d' '`
I think that support reading from stdin is  a really good idea, but having --sortTime at hand would still be good. I do not agree that our instruments necessarily have to be "orthogonal" or minimal.

Reply:
> Sorting by time from Shell is not really that easy:

I agree, all I'm arguing for here is composability via pipeline `|`

I'd be happy if you support this:

`nimgrep -r --sortTime --filenames . | nimgrep mypattern`

(the first part of the pipeline can be anything of course)
that way, we can make common useful things (eg Sorting by time) easy while still retaining full flexibility of pipelines

Reply:
Reading from stdin/pipe is implemented.
Now it's possible to do almost as you expected:
```
nimgrep -r --sortTime --filenames | nimgrep mypattern -
```
The only difference is that nimgrep reads files from current directory by default (depth 1). So to avoid a breaking change, reading from pipe requires `-`.

Reply:
`reading from pipe requires -` => good, explicit is better than implicit